samples/graphene-redis/Dockerfile

```diff
@@ -12,15 +12,26 @@ RUN make premain-graphene
 # Use with fully built graphene as build context
 # Place redis-server.manifest.template from this repo inside $GRAPHENE_DIR/Examples/redis/
 FROM ghcr.io/edgelesssys/edgelessrt-deploy:latest AS release
-RUN apt-get update && apt-get install -y build-essential autoconf gawk bison wget python3 libcurl4-openssl-dev \
-    python3-protobuf libprotobuf-c-dev protobuf-c-compiler python3-pip software-properties-common
+RUN apt-get update && apt-get install -y git meson build-essential autoconf gawk bison wget python3 libcurl4-openssl-dev \
+    python3-protobuf libprotobuf-c-dev protobuf-c-compiler python3-pip software-properties-common python3-click python3-jinja2 \
+    linux-headers-5.4.0-1049-azure
 RUN wget -qO- https://download.01.org/intel-sgx/sgx_repo/ubuntu/intel-sgx-deb.key | apt-key add
 RUN add-apt-repository 'deb [arch=amd64] https://download.01.org/intel-sgx/sgx_repo/ubuntu bionic main'
 RUN apt-get install -y libsgx-quote-ex-dev libsgx-aesm-launch-plugin
-RUN python3 -m pip install toml
-COPY . /graphene
-COPY --from=build-premain /premain/build/premain-graphene /graphene/Examples/redis
+RUN python3 -m pip install "toml>=0.10"
+
+RUN git clone https://github.com/oscarlab/graphene.git /graphene
+WORKDIR /graphene
+RUN git reset --hard b37ac75efec0c1183fd42340ce2d3e04dcfb3388
+RUN make ISGX_DRIVER_PATH=/usr/src/linux-headers-5.4.0-1049-azure/arch/x86/ SGX=1
```

Comment:
Is this ok @m1ghtym0 @daniel-weisse ?

Reply:
I'm not familiar with the best practice for this case but `git checkout b37ac75efec0c1183fd42340ce2d3e04dcfb3388` would also be an option

Reply:
Sorry, my comment was about line 26

Reply:
Just to explain the idea: I pinned the headers instead of doing `uname -r`. Those headers are needed for the AKS deployment which in my mind is the only way of deploying the container anyway. Furthermore, this way I am able to build the container outside of an SGX machine. I think another option would be to demand building the image on the same machine (e.g.  DC2s_v2) as they run on. 
Is this what you asked about?

Reply:
Isn't this gonna break badly upon a kernel update (which are common)?

Reply:
That's a good point. Since I don't know what their update policies are for their confidential aks cluster are, we should revert this back to `uname -r` ? Personally, I would much rather look up the headers once on an ACCVM and then be good to the next upgrade then to have to build the images on ACCVM itself. Is there maybe another solution? For example, remotely querying the current version?

Reply:
Can't we use the path to the SGX driver sources directly? Or ISGX_DRIVER_PATH=""? Though I am not sure the latter one works. It says it should use the in-kernel driver sources, but I am not sure if this means that only works with Linux 5.11 with the upstreamed driver or previous one. 

Nethertheless, we could specify the source directory of the isgx driver of the kernel module installed. We could get this from the list of loaded kernel modules, download it somewhere and then use this as a source directory?

Reply:
@3u13r So we're pinning it to the current Azure kernel now?
If so could you add a short disclaimer to the READMEs?

Reply:
I wanted to do that, but then @Nirusu came up with other ideas. If no one has a clear cut solution for this, I'll probably add the disclaimer, and merge this. Then either graphene changes their build or we can implement more sophisticated ideas.

Reply:
Don't like the idea of pinning this to a specific version, sorry.